initial plumbing for inspecting fevm txs

Schwartz10 opened the pull request:
use in conjuction with https://github.com/glifio/modules/pull/252

Schwartz10 committed:
initial plumbing for use abi work

use in conjuction with https://github.com/glifio/modules/pull/252

Schwartz10 committed:
Fix build issue

Schwartz10 committed:
Fix issue with fevm and no abi

Schwartz10 commented on src/components/HistoryTables/MessageHistory/Detail/MethodName.tsx:
```diff
@@ -0,0 +1,18 @@
+import { Badge, Line } from '../../../Layout'
+import { useMethodName } from '../hooks/useMethodName'
+
+export function MethodName({ address, params, methodNum }: MethodNameProps) {
```
Separating out this component fixed an issue with useAbi not updating after setting undefined as an initial value

navFooh replied:
I think this might have had something to do with SSR. Moving it into a piece of code that only renders in the browser makes sure window is defined and the right value will be set. The new useLocalStorage hook makes sure the value resaturated when going from SSR to the browser.

Schwartz10 committed:
Handle errors

Schwartz10 commented on src/components/HistoryTables/MessageHistory/hooks/useMethodName.ts:
```diff
@@ -18,6 +24,13 @@ export const useMethodName = (
+  } else if (isDelegatedAddress(address?.robust)) {
+    return 'Unknown'
+  }
+
```
Something in here errors now on the transaction history view but not the error detail view, not sure if the `abi` has something to do with this...

navFooh replied:
Might also have to do with SSR

Schwartz10 commented on src/components/Layout/DataTypes.tsx:
```diff
@@ -63,7 +63,7 @@ export const DataTypeLines = ({
 
     default:
-      throw new Error(`Unexpected DataType: ${JSON.stringify(dataType)}`)
+      return <Line label={dataType.Name}>{dataType.Value.toString()}</Line>
```
This is a hack until we figure out the data types question

Schwartz10 commented on src/components/InputV2/File.tsx:
```diff
@@ -0,0 +1,12 @@
+import { ChangeEventHandler } from 'react'
+
+export const FileInput = ({ accept, handleFileUpload }: FileInputProps) => {
+  return <input type='file' onChange={handleFileUpload} accept={accept} />
```
placeholder for a file input

Schwartz10 commented on src/utils/useAbi.ts:
```diff
@@ -0,0 +1,42 @@
+
+export type ABI = ReadonlyArray<Fragment | JsonFragment | string>
+
+export const useAbi = (actorAddr: string) => {
```
theres an annoying bug where - when you set your contract in local storage, you have to refresh the page in order for it to take effect. this is because the upload input renders its own `useAbi` hook, and doesn't share the same state with its sibling components that also render the `useAbi` hook 

there are two options i see:

1. make a context wrapper so all useAbi hooks share the same state
2. change the prop structure as to only use the useAbi hook once

I'm thinking 1 probably makes more sense?

Schwartz10 commented:
@navFooh there's a bug in here around the method name on the transaction history just FYI

Schwartz10 committed:
useAbi hook improvements (#614)

* useAbi hook improvements

* Cleanup

navFooh committed:
Update File input

navFooh committed:
Update Abi.tsx

navFooh committed:
Allow to customize drop-text

navFooh committed:
Merge branch 'primary' into useAbi

navFooh committed:
Update useLocalStorage

navFooh committed:
Merge branch 'primary' into useAbi

navFooh committed:
install uint8arrays

navFooh committed:
revert method name

navFooh committed:
Move ETH address

navFooh committed:
Update File.tsx

navFooh committed:
Fix button line-height

navFooh committed:
Add ABI line to actor state

navFooh committed:
Update address checks

navFooh committed:
Update useMethodName.ts

navFooh committed:
Update index.tsx

navFooh committed:
Update index.ts

navFooh committed:
update useEnvironment import path

navFooh committed:
Update useMethodName.ts

navFooh committed:
Merge branch 'primary' into useAbi

navFooh committed:
Update parsing params and returns

navFooh committed:
Fix GasCharges proptype error

navFooh committed:
Update actor utils

Update detail.tsx

Update DataTypes.tsx

Update detail.tsx

Update DataTypes.tsx

update @glif/filecoin-actor-utils

navFooh committed:
Update snapshots

navFooh committed:
Revert to calibrationnet

navFooh committed:
Update snapshots

navFooh committed:
revert actor address

navFooh committed:
Revert message address

navFooh committed:
Remove extra space

navFooh committed:
Revert history address

navFooh committed:
Update message detail layout

Update detail.tsx

layout

Update index.test.tsx.snap
